### PR TITLE
[wallet] Fixups from account API deprecation

### PR DESCRIPTION
**Pull request by jnewbery**
A couple of fixups from the accounts API deprecation PR (#12953):

- properly deprecate `sendfrom`
- don't use accounts when calculating balance in `sendmany` (unless the `-deprecatedrpc=accounts` flag is being used)

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -1255,9 +1263,11 @@ static UniValue sendmany(const JSONRPCRequest& request)
     EnsureWalletIsUnlocked(pwallet);
 
     // Check funds
-    CAmount nBalance = pwallet->GetLegacyBalance(ISMINE_SPENDABLE, nMinDepth, &strAccount);
-    if (totalAmount > nBalance)
+    if (IsDeprecatedRPCEnabled("accounts") && totalAmount > pwallet->GetLegacyBalance(ISMINE_SPENDABLE, nMinDepth, &strAccount)) {
         throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS, "Account has insufficient funds");
+    } else if (!IsDeprecatedRPCEnabled("accounts") && totalAmount > pwallet->GetLegacyBalance(ISMINE_SPENDABLE, nMinDepth, nullptr)) {
+        throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS, "Wallet has insufficient funds");
```
[the remark this answers is not in the document]
Comment (sipa) [reply src/wallet/rpcwallet.cpp]: The label in sendfrom/sendmany denotes which account's balance to debit. It does not, and never did, affect which UTXO were being used in transaction creation.

[the remark this answers is not in the document]
Comment (laanwj) [reply src/wallet/rpcwallet.cpp]: > The label in sendfrom/sendmany denotes which account's balance to debit. It does not, and never did, affect which UTXO were being used in transaction creation.

Indeed, this is one of the biggest confusions around accounts, and why they're being deprecated.

[the remark this answers is not in the document]
Comment (TheCharlatan) [reply src/wallet/rpcwallet.cpp]: Yes, both sendfrom and sendmany use the same api of `CreateTransaction` (where no direct coin control is used in this case) followed by `CommitTransaction`. I think the wording would be better if "send" from is changed with "debited" from (both in the api call and in my comment, but that is probably not possible anymore). 
Currently I have 2 UTXOs with 2 confirmations each in my wallet. One has an address with the label "THIS" and a balance of 0.19 BTC, the other is unlabeled and has 0.009 BTC. If I `getbalance` the amount is displayed correctly as 0.199BTC. If I now to try to `sendmany "" '{"someaddress":0.03}'`, the `Account has insufficient funds (code -6)` error is thrown. Naively, I would expect this to work.

[the remark this answers is not in the document]
Comment (jnewbery) [reply src/wallet/rpcwallet.cpp]: @TheCharlatan - please see my comment here: https://github.com/bitcoin/bitcoin/issues/12952#issuecomment-398103755.

The account is not used in coin selection, but it *is* used in calculating the balance and whether there are enough funds to send from. That's what this PR is fixing.

[the remark this answers is not in the document]
Comment (TheCharlatan) [reply src/wallet/rpcwallet.cpp]: Sorry for the ruckus, this patch does exactly what it should do.

[the remark this answers is not in the document]
Comment (jnewbery) [reply src/wallet/rpcwallet.cpp]: @TheCharlatan no problem. Any test/review of this PR would be greatly appreciated :slightly_smiling_face:

**Review by sedited — APPROVED** [review]
Tested sendmany and sendfrom, seem to work as intended.

**Review by meshcollider — COMMENTED** [review]
utACK https://github.com/bitcoin/bitcoin/commit/df10f07db178d84b3def9f0b99d9b7c92e4227a9

**Comment by sipa** [comment]
Concept ACK

**Comment by laanwj** [comment]
utACK df10f07db178d84b3def9f0b99d9b7c92e4227a9

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (promag) [inline src/wallet/rpcwallet.cpp]: Commit "[wallet] Don't use accounts when checking balance in sendmany"

Use same error as https://github.com/bitcoin/bitcoin/blob/ee02debb25675542d13a03b65ed49832b79b0db9/src/wallet/rpcwallet.cpp#L466
and
https://github.com/bitcoin/bitcoin/blob/ee02debb25675542d13a03b65ed49832b79b0db9/src/wallet/wallet.cpp#L2905

**Review by promag — COMMENTED** [review]
utACK df10f07. One comment though, feel free to ignore since the error code is more important.

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (jnewbery) [reply src/wallet/rpcwallet.cpp]: Thanks @promag . This code will need to be changed for V0.18 when accounts are definitively removed. Let's save error message cleanup until then and not invalidate previous ACKs.

**Pull request merged**